packages/protocol/simulation/simulation_data_parser.py

```diff
@@ -0,0 +1,150 @@
+            if HIGH_TRAFFIC_ENDS in line:
+                 # Saving the first plot
+                x_axis = x_ax_block_nr
+                y_axis = y_ax_base_fee
+                filename = "./plots/{}_above_average_traffic.png".format(timestamp)
+                title = "Basefee chart per L2 block (if gas used above target)"
+                y_label = "baseFee"
+                x_label = "L2 block count"
+                y_signal_label = "basefee in gwei"
+                save_plots(x_axis, y_axis, filename, title, x_label, y_label, y_signal_label)
+                x_ax_block_nr = []
+                y_ax_base_fee = []
+                PARSER_MODE = 0
+                continue
+            
+            if LOW_TRAFFIC_STARTS in line:
+                PARSER_MODE = 2
+                continue
+            
+            if LOW_TRAFFIC_ENDS in line:
+                 # Saving the first plot
+                x_axis = x_ax_block_nr
+                y_axis = y_ax_base_fee
+                filename = "./plots/{}_below_average_traffic.png".format(timestamp)
+                title = "Basefee chart per L2 block (if gas used below target)"
+                y_label = "baseFee"
+                x_label = "L2 block count"
+                y_signal_label = "basefee in gwei"
+                save_plots(x_axis, y_axis, filename, title, x_label, y_label, y_signal_label)
+                x_ax_block_nr = []
+                y_ax_base_fee = []
+                PARSER_MODE = 0
+                continue
+
+            if TARGET_TRAFFIC_STARTS in line:
+                PARSER_MODE = 3
+                continue
+            
+            if TARGET_TRAFFIC_ENDS in line:
+                 # Saving the first plot
+                x_axis = x_ax_block_nr
+                y_axis = y_ax_base_fee
+                filename = "./plots/{}_target_traffic.png".format(timestamp)
+                title = "Basefee chart per L2 block (if gas used around target)"
+                y_label = "baseFee"
+                x_label = "L2 block count"
+                y_signal_label = "basefee in gwei"
+                save_plots(x_axis, y_axis, filename, title, x_label, y_label, y_signal_label)
+                x_ax_block_nr = []
+                y_ax_base_fee = []
+                PARSER_MODE = 0
+                continue
+            
+            if INFO in line and PARSER_MODE != 0:
+                data = line.rstrip().split(':')
+                x_ax_block_nr.append(float(data[1]))
+                y_ax_base_fee.append(float(data[2]) / 1000000000)
+                continue
+
+            if AVERAGE_GAS_USED_PER_L1 in line:
+                data = line.rstrip().split(':')
+                AVG_GAS_USED_PER_L1_BLOCK = float(data[1]) / 1000000
+                continue
+
+            if AVERAGE_GAS_PRICE_IN_L2 in line:
+                data = line.rstrip().split(':')
+                AVG_BASEFEE_PER_L2_BLOCK = float(data[1]) / 1000000000
+                continue
+
+def parse_args():
```

Comment:
can we have default parameter values at the top of the file so we can easily modify there?

Reply:
This is just a parser and plotter. Please modify `TaikoL2.t.sol`

Reply:
I try to run `python3 simulation_data_parser.py ./exports/simulation_data_XXXXXXXXXX.txt` but end up with error :

```
Traceback (most recent call last):
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 150, in <module>
    main()
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 146, in main
    parse_and_plot(inputs.file)
  File "/Users/d/Projects/taiko/taiko-mono/packages/protocol/simulation/simulation_data_parser.py", line 62, in parse_and_plot
    with open(filename) as file:
```

Reply:
Replace `XXXX`.